g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationWithPatternAction.java

```diff
@@ -51,6 +53,12 @@
 	    */
 	   public APIResponseDTO getTransPattern(TranslationWithPatternDTO data) throws Exception {
 	      logger.info("begin getTransPattern");
+	      try {
+	    	  ParameterValidationUtility.validateTranslationWithPatternAPI(data);
+	      }catch(VIPAPIException excep) {
+	    	  return super.handleResponse(APIResponseStatus.BAD_REQUEST.getCode(),
+	    			  excep.getMessage(), null);
+	      }
```

Comment:
1. put the validation in 'validate' function
2. don't need the catch since the function will throw it.

Reply:
2, if don't catch the exception, the default http response code is 520